src/librustc_lint/nonstandard_taste.rs

```diff
@@ -0,0 +1,52 @@
+use crate::{LateContext, LateLintPass, LintContext};
+use rustc_hir as hir;
+
+declare_lint! {
+    pub PINEAPPLE_ON_PIZZA,
+    Forbid,
```

Comment:
I don't think this needs its own lint group. It should go with other similar lints, like `allow(bad_style)`.

Reply:
Not `allow(bad_style)`, but `allow(bad_taste)`.

Reply:
I don't think we should allow pineapple pizza to be instantiate regardless of the pragma or directive. We shouldn't permit terrorism.